src/util.rs

```diff
@@ -1,38 +1,26 @@
-/// HACK! To reduce code duplication in the generated binary, this macro is used to generate a
+/// HACK! To reduce code duplication in the generated binary, this function is used to generate a
```

Comment:
> code duplication in the generated binary

I don't think this reduces code duplication in the binary, right? Since the struct is a generic this does reduce the amount of source code written, but I don't think it has any impact on the generated code

Reply:
I clarified the meaning of the comment. If we made the functions in the `private` module generic, it would lead to code duplication, is what I meant